Lib/functools.py

```diff
@@ -447,54 +442,94 @@ class partialmethod:
     Supports wrapping existing descriptors and handles non-descriptor
     callables as instance methods.
     """
-    __new__ = _partial_new
+
+    __slots__ = ("func", "args", "keywords", "wrapper",
+                 "__dict__", "__weakref__")
+
     __repr__ = _partial_repr
 
-    def _make_unbound_method(self):
-        def _method(cls_or_self, /, *args, **keywords):
-            phcount = self._phcount
-            if phcount:
-                try:
-                    pto_args = self._merger(self.args + args)
-                    args = args[phcount:]
-                except IndexError:
-                    raise TypeError("missing positional arguments "
-                                    "in 'partialmethod' call; expected "
-                                    f"at least {phcount}, got {len(args)}")
-            else:
-                pto_args = self.args
-            keywords = {**self.keywords, **keywords}
-            return self.func(cls_or_self, *pto_args, *args, **keywords)
-        _method.__isabstractmethod__ = self.__isabstractmethod__
-        _method.__partialmethod__ = self
-        return _method
+    def __init__(self, func, /, *args, **keywords):
+        if isinstance(func, partialmethod):
+            # Subclass optimization
+            temp = partial(lambda: None, *func.args, **func.keywords)
+            temp = partial(temp, *args, **keywords)
```

Comment:
So this is just to get the correct "args" and "keywords" right? But it's a bit weird to see that `lambda: None` is being used with `func.args` and `func.keywords`. Why not doing:

```py
temp = partial(lambda *_, **__: None, *func.args, *args, **func.keywords, **keywords)
```